src/snowflake/snowpark/modin/pandas/groupby.py

```diff
@@ -602,6 +608,22 @@ def aggregate(
             how="axis_wise",
             is_result_dataframe=is_result_dataframe,
         )
+
+        if uses_named_agg:
+            # When we use NamedAggregations, the output columns must be in the same order
+            # as they were passed in via the kwargs, but due to how we process the NamedAggregations
+            # they can end up out of order, as named aggregations on the same column will be moved to
+            # be contiguous.
+            # Example: say we have new_col1=('A', 'max'), new_col2=('B', 'min'), new_col3=('A', 'min')
+            # Our result will have columns ordered like so: [new_col1, new_col3, new_col2]. We need
```

Comment:
why we would have new_col1, new_col3, and new_col2, i assume if you have a mapping entered, the order should be expected?
Also can we try to do the indexing correctly at query compiler groupby_agg layer, since we do the ordering preserving all at that layer today, and we want to have most of the functionality happen at query compiler layer with the modin upstreaming work also

Reply:
Sounds good - added a comment explaining why this happens, and moved the reindexing to the qc layer!